speechbrain/core.py

```diff
@@ -1151,6 +1163,21 @@ def _fit_train(self, train_set, epoch, enable):
         self.step = 0
         self.valid_step = 0
 
+    def _should_save_intra_epoch_ckpt(self, last_ckpt_time, steps_since_ckpt):
+        """Determines if an intra-epoch checkpoint should be saved.
+
+        Returns True if there's a checkpointer and time or steps has exceeded limit.
+        """
+        if self.checkpointer is None:
+            return False
+
+        # Check if we've run for the requested amount of time
+        if 0 < self.ckpt_interval_minutes * 60.0 < time.time() - last_ckpt_time:
+            return True
```

Comment:
I still think here we would want to check if we are on the main process. In the current approach we are letting all processes call the checkpointer at different times and just expecting the checkpointer to work right. I think this invites bugs where users think they can rely on the different processes calling the checkpointer at the same time.

Reply:
Okay, I've added this, and updated the warning accordingly. Do you think the warning is still needed or should we remove it altogether?